### PR TITLE
Change Ska.engarchive to cheta w/ back-compatibility

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

This is the realization of a years-long dream to change Ska.engarchive to cheta for real.

## Interface impacts

No community user-facing changes.

The location of the `task_schedule.cfg` file is moved to within the package, requiring a change to the cron job command. This is captured as an install process step. 

The previously-installed `task_schedule_gap.cfg` is now only available in the git repo. This task schedule config is no longer used in practice because gap recovery is built into the update code.

### Unit tests

- [x] Mac

Independent check of unit tests by Jean.  cheta and Ska.engarchive tests pass for me in an environment with conda installed 'masters' versions of cxotime, ska_helpers, kadi.
- [x] Linux

#### Installation and package file manifest

From the cheta git repo in a dev ska3 environment `ska3-dev`:
```
$ python setup.py install --single-version-externally-managed --record=record.txt
```
The install manifest is below, sans the root install path:
```
lib/python3.8/site-packages/cheta/fetch.py
lib/python3.8/site-packages/cheta/update_server_sync.py
lib/python3.8/site-packages/cheta/fetch_eng.py
lib/python3.8/site-packages/cheta/plot.py
lib/python3.8/site-packages/cheta/update_client_archive.py
lib/python3.8/site-packages/cheta/units.py
lib/python3.8/site-packages/cheta/add_derived.py
lib/python3.8/site-packages/cheta/file_defs.py
lib/python3.8/site-packages/cheta/fetch_sci.py
lib/python3.8/site-packages/cheta/get_telem.py
lib/python3.8/site-packages/cheta/cache.py
lib/python3.8/site-packages/cheta/__init__.py
lib/python3.8/site-packages/cheta/fix_bad_values.py
lib/python3.8/site-packages/cheta/remote_access.py
lib/python3.8/site-packages/cheta/update_archive.py
lib/python3.8/site-packages/cheta/utils.py
lib/python3.8/site-packages/cheta/fetch_old.py
lib/python3.8/site-packages/cheta/check_integrity.py
lib/python3.8/site-packages/cheta/converters.py
lib/python3.8/site-packages/cheta/lazy.py
lib/python3.8/site-packages/cheta/derived/thermal.py
lib/python3.8/site-packages/cheta/derived/orbit.py
lib/python3.8/site-packages/cheta/derived/mups_valve.py
lib/python3.8/site-packages/cheta/derived/__init__.py
lib/python3.8/site-packages/cheta/derived/test.py
lib/python3.8/site-packages/cheta/derived/acispow.py
lib/python3.8/site-packages/cheta/derived/pcad.py
lib/python3.8/site-packages/cheta/derived/eps.py
lib/python3.8/site-packages/cheta/derived/comps.py
lib/python3.8/site-packages/cheta/derived/base.py
lib/python3.8/site-packages/cheta/tests/test_utils.py
lib/python3.8/site-packages/cheta/tests/test_sync.py
lib/python3.8/site-packages/cheta/tests/test_data_source.py
lib/python3.8/site-packages/cheta/tests/__init__.py
lib/python3.8/site-packages/cheta/tests/test_comps.py
lib/python3.8/site-packages/cheta/tests/test_fetch.py
lib/python3.8/site-packages/cheta/tests/test_units.py
lib/python3.8/site-packages/cheta/tests/test_orbit.py
lib/python3.8/site-packages/cheta/tests/test_remote_access.py
lib/python3.8/site-packages/cheta/tests/test_intervals.py
lib/python3.8/site-packages/cheta/tests/test_units_reversed.py
lib/python3.8/site-packages/Ska/engarchive/fetch.py
lib/python3.8/site-packages/Ska/engarchive/update_server_sync.py
lib/python3.8/site-packages/Ska/engarchive/fetch_eng.py
lib/python3.8/site-packages/Ska/engarchive/plot.py
lib/python3.8/site-packages/Ska/engarchive/update_client_archive.py
lib/python3.8/site-packages/Ska/engarchive/units.py
lib/python3.8/site-packages/Ska/engarchive/add_derived.py
lib/python3.8/site-packages/Ska/engarchive/file_defs.py
lib/python3.8/site-packages/Ska/engarchive/fetch_sci.py
lib/python3.8/site-packages/Ska/engarchive/get_telem.py
lib/python3.8/site-packages/Ska/engarchive/cache.py
lib/python3.8/site-packages/Ska/engarchive/__init__.py
lib/python3.8/site-packages/Ska/engarchive/fix_bad_values.py
lib/python3.8/site-packages/Ska/engarchive/remote_access.py
lib/python3.8/site-packages/Ska/engarchive/update_archive.py
lib/python3.8/site-packages/Ska/engarchive/utils.py
lib/python3.8/site-packages/Ska/engarchive/fetch_old.py
lib/python3.8/site-packages/Ska/engarchive/check_integrity.py
lib/python3.8/site-packages/Ska/engarchive/converters.py
lib/python3.8/site-packages/Ska/engarchive/lazy.py
lib/python3.8/site-packages/Ska/engarchive/derived/thermal.py
lib/python3.8/site-packages/Ska/engarchive/derived/orbit.py
lib/python3.8/site-packages/Ska/engarchive/derived/mups_valve.py
lib/python3.8/site-packages/Ska/engarchive/derived/__init__.py
lib/python3.8/site-packages/Ska/engarchive/derived/test.py
lib/python3.8/site-packages/Ska/engarchive/derived/acispow.py
lib/python3.8/site-packages/Ska/engarchive/derived/pcad.py
lib/python3.8/site-packages/Ska/engarchive/derived/eps.py
lib/python3.8/site-packages/Ska/engarchive/derived/comps.py
lib/python3.8/site-packages/Ska/engarchive/derived/base.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_utils.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_sync.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_data_source.py
lib/python3.8/site-packages/Ska/engarchive/tests/__init__.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_comps.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_fetch.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_units.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_orbit.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_remote_access.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_intervals.py
lib/python3.8/site-packages/Ska/engarchive/tests/test_units_reversed.py
lib/python3.8/site-packages/cheta/task_schedule.cfg
lib/python3.8/site-packages/cheta/filetypes.dat
lib/python3.8/site-packages/cheta/msid_bad_times.dat
lib/python3.8/site-packages/cheta/units_cxc.pkl
lib/python3.8/site-packages/cheta/units_sci.pkl
lib/python3.8/site-packages/cheta/units_eng.pkl
lib/python3.8/site-packages/cheta/archfiles_def.sql
lib/python3.8/site-packages/cheta/tests/fetch_regr.dat
lib/python3.8/site-packages/Ska/engarchive/task_schedule.cfg
lib/python3.8/site-packages/Ska/engarchive/filetypes.dat
lib/python3.8/site-packages/Ska/engarchive/msid_bad_times.dat
lib/python3.8/site-packages/Ska/engarchive/units_cxc.pkl
lib/python3.8/site-packages/Ska/engarchive/units_sci.pkl
lib/python3.8/site-packages/Ska/engarchive/units_eng.pkl
lib/python3.8/site-packages/Ska/engarchive/archfiles_def.sql
lib/python3.8/site-packages/Ska/engarchive/tests/fetch_regr.dat
lib/python3.8/site-packages/cheta-4.57.1.dev24+g3a9214d.d20221221-py3.8.egg-info/PKG-INFO
lib/python3.8/site-packages/cheta-4.57.1.dev24+g3a9214d.d20221221-py3.8.egg-info/not-zip-safe
lib/python3.8/site-packages/cheta-4.57.1.dev24+g3a9214d.d20221221-py3.8.egg-info/SOURCES.txt
lib/python3.8/site-packages/cheta-4.57.1.dev24+g3a9214d.d20221221-py3.8.egg-info/entry_points.txt
lib/python3.8/site-packages/cheta-4.57.1.dev24+g3a9214d.d20221221-py3.8.egg-info/top_level.txt
lib/python3.8/site-packages/cheta-4.57.1.dev24+g3a9214d.d20221221-py3.8.egg-info/dependency_links.txt
bin/cheta_add_derived
bin/cheta_check_integrity
bin/cheta_fix_bad_values
bin/cheta_sync
bin/cheta_update_server_archive
bin/cheta_update_server_sync
bin/ska_fetch
```
#### Testing
```
$ cd docs  # so cheta in git repo is not in path
$ ipython
(ska3-dev) ➜  docs git:(de-namespace) ipython
Python 3.8.12 (default, Oct 12 2021, 06:23:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.29.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import Ska.engarchive

In [2]: Ska.engarchive.__file__
Out[2]: '/Users/aldcroft/miniconda3/envs/ska3-dev/lib/python3.8/site-packages/Ska/engarchive/__init__.py'

In [3]: Ska.engarchive.__version__
Out[3]: '4.57.1.dev25+g83f97e2'

In [4]: Ska.engarchive.test()
....SUCCESS...
```
```
$ipython
(ska3-dev) ➜  docs git:(de-namespace) ipython
Python 3.8.12 (default, Oct 12 2021, 06:23:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.29.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import cheta

In [2]: cheta.__file__
Out[2]: '/Users/aldcroft/miniconda3/envs/ska3-dev/lib/python3.8/site-packages/cheta/__init__.py'

In [3]: cheta.__version__
Out[3]: '4.57.1.dev25+g83f97e2'

In [4]: cheta.test()
... SUCCESS
```

#### Note
~~Full disclosure: running both the `Ska.engarchive` tests and then the `cheta` tests in the same session failed~~

Fixed by 51bfc77

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,40 +12,32 @@
...
     author_email="taldcroft@cfa.harvard.edu",
     entry_points={"console_scripts": console_scripts},
     use_scm_version=True,
     setup_requires=["setuptools_scm", "setuptools_scm_git_archive"],
     zip_safe=False,
-    package_dir={"Ska": "Ska", "cheta": "Ska/engarchive"},
+    package_dir=package_dir,
     packages=packages,
     package_data=package_data,
-    data_files=data_files,
```
Comment (jeanconn) [inline setup.py:56]: It should probably be called out as an interface change that the task_schedule.cfg has moved and the task_schedule_gap.cfg is no longer installed. (And I don't know if we need to take a post-install step to remove the old ones).

**Comment by jeanconn** [comment]
My test install shows
```
warning: no files found matching 'Ska/engarchive/units_*.pkl'
warning: no files found matching 'Ska/engarchive/*.dat'
warning: no files found matching 'Ska/engarchive/tests/*.dat'
warning: no files found matching 'Makefile'
warning: no files found matching 'update_archive.py'
warning: no files found matching 'transfer_stage.py'
warning: no files found matching 'archfiles_def.sql'
```
So I think the manifest needs update.

File: setup.py (hunk 1 above)
Comment (taldcroft) [reply setup.py:56]: Done, with updates to the description and a post-install step added to https://github.com/sot/skare3/issues/976.

**Comment by taldcroft** [comment]
About the warnings related to `MANIFEST.in`, several of those warnings had already been present in master (I confirmed this with an install test from master). For instance `update_archive.py` and `Makefile` are no longer there at the top level and generate warnings.

 So I just removed that file and re-ran the same install + unit tests shown above on Mac. All PASS.

**Comment by jeanconn** [comment]
Sounds good regarding MANIFEST(.in) and those missing files not being a regression.  I was just looking pretty closely at install behaviors with this PR.

**Review by jeanconn — APPROVED** [review]

**Pull request merged**